src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+        self.output_key = output_key
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Augment the input record with the computed faithfulness score.
+
+        :param record: The input record.
+        :returns: The input record, with the faithfulness score added in.
+        """
+        verdict_output = record["raw_verdicts"]
+        statements = record["statements"]
+        record[self.output_key] = self._get_score(verdict_output, statements)
+        return record
+
+    @staticmethod
+    def _get_score(verdict_output: str, statements: str) -> float:
+        """Given generated statements and verdicts, compute Faithfulness score.
+
+        :param verdict_output: Verdicts(Yes/No) and explanations string get from Judge model.
+        :param statements: Statements string get from `GetStatements` Transform.
+        :returns: 0 to 1. See the docstring for `Faithfulness` for more details
+            on what these numerical values represent.
+        """
+        output = verdict_output.lower().strip()
+        num_statements = len(statements.split("\n"))
+        # TODO: handle edge case that num_statements is 0?
+
+        # calculate score
+        final_answer = "Final verdicts in order:"
+        final_answer = final_answer.lower()
+        if output.find(final_answer) != -1:
+            output = output[output.find(final_answer) + len(final_answer) :]
+            score = float(sum(1 if "yes" in answer else 0 for answer in output.strip().split(".") if answer != ""))
+            score = score / num_statements
+        else:
+            score = float(max(0, output.count("verdict: yes")) / num_statements)
+
+        return score
+
+
+class GetStatements(Transform):
+    """This transform invokes the judge model string from raw statements."""
+
+    def __init__(self, input_key: str, output_key: str, judge_model: ModelRunner):
+        """GetStatements Initializer.
+        :param input_key: The key corresponding to prompt to get statements.
+        :param output_key: The key corresponding to the statements, which gets added to the record.
+        :param judge_model: An instance of ModelRunner representing the judge model to be used.
+            If this argument is None, default judge model will be provided.
+        """
+        super().__init__(input_key, output_key, judge_model)
+        self.register_input_output_keys(
+            input_keys=[input_key],
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and arguments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the mean added in.
+        """
+        get_raw_statements = GetModelOutputs(
```

Comment:
If you are using a `Transform` (`GetModelOutputs`) within a `Transform` (`GetStatements`), could this be avoided by just adding this `GetModelOutputs` as a previous step?

Reply:
Yes this can be. I initially put them separately, but another transform step to process the raw model response is needed. So I feel it make more sense to put them together as one step.